tests/test_area_detector.py

```diff
@@ -1,165 +1,214 @@
+    assert True is await drv.wait_for_plugins.get_value()
 
+    assert d.names == ["start", "descriptor", "event", "stop"]
+    _, descriptor, event, _ = d.docs
+    assert descriptor["configuration"]["det"]["data"]["det-drv-acquire_time"] == 0.5
+    assert event["data"]["det-drv-array_counter"] == 1
+    assert event["data"]["det-stats-unique_id"] == 3
 
-async def test_hdf_writing_sim(hdf_writing_sim: MyHDFWritingSim, RE):
-    det = hdf_writing_sim
-    set_sim_value(det.drv.acquire, 0)
-    set_sim_value(det.drv.image_mode, ImageMode.multiple)
-    set_sim_value(det.drv.wait_for_plugins, False)
-    set_sim_value(det.hdf.lazy_open, False)
-    set_sim_value(det.hdf.file_write_mode, FileWriteMode.single)
-
-    docs = []
-
-    def append_callback(*args, **kwargs):
-        docs.append(args)
-
-    RE(bp.count([det]), append_callback)
-
-    assert (await det.drv.acquire.get_value()) == 1
-    assert (await det.drv.image_mode.get_value()) == ImageMode.single
-    assert await det.drv.wait_for_plugins.get_value()
-    assert await det.hdf.lazy_open.get_value()
-    assert (await det.hdf.file_write_mode.get_value()) == FileWriteMode.stream
-
-    _, (_, descriptor), (_, resourse), (_, datum), _, _ = docs
-
-    assert (
-        descriptor["configuration"]["hdf_writing_sim"]["data"][
-            "hdf_writing_sim-drv-acquire_time"
-        ]
-        == 0.5
-    )
-    assert resourse["resource_path"] == "/tmp/tempfile"
-    assert datum["datum_kwargs"]["point_number"] == 1
-
-
-async def test_hdf_flyer_sim(hdf_flyer_sim: MyHDFFlyerSim, RE):
-    det = hdf_flyer_sim
-    set_sim_value(det.drv.num_images, 0)
-    set_sim_value(det.drv.acquire, 0)
-    set_sim_value(det.drv.image_mode, ImageMode.single)
-    set_sim_value(det.drv.wait_for_plugins, False)
-    set_sim_value(det.hdf.lazy_open, False)
-    set_sim_value(det.hdf.file_write_mode, FileWriteMode.single)
-
-    docs = []
-
-    def append_callback(*args, **kwargs):
-        docs.append(args)
 
+@pytest.fixture
+async def hdf_streamer_dets():
+    namer = HDFNamer()
+    async with DeviceCollector(sim=True):
+        deta = HDFStreamerDet(
+            drv=ADDriver(prefix="PREFIX1:DET"),
+            hdf=NDFileHDF("PREFIX1:HDF"),
+            namer=namer,
+        )
+        detb = HDFStreamerDet(
+            drv=ADDriver(prefix="PREFIX1:DET"),
+            hdf=NDFileHDF("PREFIX1:HDF"),
+            namer=namer,
+        )
+
+    assert deta.name == "deta"
+    assert detb.name == "detb"
+    assert deta.drv.name == "deta-drv"
+    assert deta.hdf.name == "deta-hdf"
+
+    for det in (deta, detb):
+        set_sim_value(det.drv.acquire_time, 0.8)
+        set_sim_value(det.drv.image_mode, ImageMode.continuous)
+        set_sim_value(det.hdf.num_capture, 1000)
+        set_sim_value(det.hdf.num_captured, 1)
+        set_sim_value(det.hdf.full_file_name, f"/tmp/123456/{det.name}.h5")
+        set_sim_value(det.drv.array_size_x, 1024)
+        set_sim_value(det.drv.array_size_y, 768)
```

Comment:
same here as above. We can replace some of
this, and maybe think about not even setting the readable values:

```python
await det.drv.acquire_time.set(0.8)
await det.drv.image_mode.set(ImageMode.continuous)
await det.hdf.num_capture.set(1000)

# set read only values
set_sim_value(det.hdf.num_captured, 1)
set_sim_value(det.hdf.full_file_name, f"/tmp/123456/{det.name}.h5")
set_sim_value(det.drv.array_size_x, 1024)
set_sim_value(det.drv.array_size_y, 768)
```

I'm not even sure the last two are actually used. If I comment them out the tests still pass.

Reply:
They come out in the descriptor, but maybe I didn't actually check that in the tests, will add that check too

Reply:
yep adding this check sounds like a good idea. Am happy for you to leave these as set_sim_value with an appropriate comment as you've explained above :+1: